src/pages/ShufflePlayer.js

```diff
@@ -7,76 +7,70 @@ import Player from '../components/Player';
 import PlaylistSelector from '../components/PlaylistSelector';
 import PlayedHistory from '../components/PlayedHistory';
 import CurrentVideoInfo from '../components/CurrentVideoInfo';
+import useVideoHook from '../hooks/VideoHook';
 
-function ShufflePlayer(props) {
-  const [isLoaded, setIsLoaded] = useState(false);
+function randomInteger(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function ShufflePlayer() {
   const [loadedPlaylists, setLoadedPlaylists] = useState([]);
-  const [loadedVideos, setLoadedVideos] = useState([]);
   const [currentVideo, setCurrentVideo] = useState({});
   const [playedVideos, setPlayedVideos] = useState([]);
   const [repeatVideo, setRepeatVideo] = useState(false);
-  const [hideVideo, setHideVideo] = useState(false);
+  const [hideVideo, setHideVideo] = useState(false);  
+  const [playlistIds, setPlaylistIds] = useState([]);
+  const [videoResult, reloadVideos] = useVideoHook(playlistIds);
 
   function loadPlaylists() {
     axios.get(AppConstants.APIEndpoints.TRACKED_PLAYLISTS)
     .then(response => {
       setLoadedPlaylists(response.data);
-      setIsLoaded(true);
     })
     .catch(error => console.log(`Couldn't retrieve tracked playlists! ${error}`))
   }
 
-  function loadVideos() {
-    const playlistIdsToLoad = loadedPlaylists
-      .filter(p => p.is_default)
-      .map(p => p.playlist_id);
-    const requestBody = { playlist_ids: playlistIdsToLoad };
-    axios.post(AppConstants.APIEndpoints.SHUFFLE, requestBody)
-    .then(response => setLoadedVideos(response.data.songs))
-    .catch(error => console.log(`Couldn't retrieve playlist videos! ${error}`))
-  }
-
   function togglePlaylistSelection(togglePlaylistId) {
     const toggledOnePlaylist = loadedPlaylists.map(p => { 
       return p.playlist_id === togglePlaylistId ? 
         { ...p, is_default: !p.is_default } : p});
+    const playlistIds = toggledOnePlaylist
```

Comment:
Wait is this supposed to say toggledOn ? I never noticed that before lmao;
we can fix the spelling in a follow up to avoid having to deal with more changes in this pr